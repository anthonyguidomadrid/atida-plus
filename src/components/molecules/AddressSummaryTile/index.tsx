export * from './AddressSummaryTile'
export * from './AddressSummaryTile.mock'
