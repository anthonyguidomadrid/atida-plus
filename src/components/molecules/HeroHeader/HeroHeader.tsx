import { ComponentPropsWithoutRef, FunctionComponent, memo } from 'react'
import { useTranslation } from 'react-i18next'

import { ContentBlocksLayout } from '~components/templates/ContentBlocksLayout'
import { CampaignHeroBanner, HeroBanner } from '~domains/contentful'
import { ReactComponent as ChevronLeft } from '~assets/svg/navigation-16px/ChevronLeft.svg'
import { Button } from '~components/atoms/Button'

export type HeroHeaderProps = {
  contentBlock: HeroBanner | CampaignHeroBanner
  backFunction: () => void
}

const HeroHeaderComponent: FunctionComponent<
  ComponentPropsWithoutRef<'div'> & HeroHeaderProps
> = ({ contentBlock, backFunction }) => {
  const { t } = useTranslation()

  return (
    <div className="relative" data-testid="heroHeader">
      <Button
        variant="back"
        icon={<ChevronLeft role="presentation" className="icon-16" />}
        onClick={backFunction}
        data-testid="popBackButton"
        className="absolute z-2 ml-3 sm:ml-6 lg:ml-8 top-2 sm:top-3"
      >
        {t('shared.back')}
      </Button>

      <ContentBlocksLayout
        contentBlocks={[contentBlock]}
        isContainer={false}
        isLcp={true}
      />
    </div>
  )
}

export const HeroHeader = memo(HeroHeaderComponent)
