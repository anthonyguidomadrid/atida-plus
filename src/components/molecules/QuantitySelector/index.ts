export * from './QuantitySelector'
export * from './ProductQuantitySelector'
