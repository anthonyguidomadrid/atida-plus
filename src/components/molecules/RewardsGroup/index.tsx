export * from './RewardsGrid'
export * from './RewardsList'
export * from './RewardsGroupHelpers'
export * from './__mocks__/RewardsGroup'
