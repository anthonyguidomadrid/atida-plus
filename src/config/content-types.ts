export enum CmsContentTypes {
  CONTENT_BLOCK_WITH_IMAGE = 'ContentBlockWithImage',
  SLIDER = 'Slider',
  HERO_BANNER = 'HeroBanner',
  CAMPAIGN_HERO_BANNER = 'CampaignHeroBanner',
  CATEGORY_GRID = 'CategoryGrid',
  TOP_BRANDS = 'TopBrands',
  TRANSLATIONS = 'Translations',
  USPS_CARD = 'UspsCard',
  SEO = 'Seo',
  LINK_BLOCK = 'LinkBlock',
  PROMOTION = 'Promotion',
  STATIC_CONTENT_BLOCK = 'StaticContentBlock',
  STATIC_HEADER_BLOCK = 'StaticHeaderBlock',
  EXPONEA_RECOMMENDATION = 'ExponeaRecommendation',
  CONTAINER_OF_CONTENT_BLOCKS = 'ContainerOfContentBlocks',
  GROUP_OF_STATIC_CONTENT_BLOCKS = 'GroupOfStaticContentBlocks',
  VOUCHER_CODES = 'VoucherCodes',
  STATIC_RECOMMENDATION_BLOCK = 'StaticRecommendationBlock',
  EXPERT_SIGNATURE = 'ExpertSignature'
}

export enum CmsContainerContentTypes {
  PROMOTION = 'Promotion',
  MARKETING_TEASER = 'MarketingTeaser'
}
