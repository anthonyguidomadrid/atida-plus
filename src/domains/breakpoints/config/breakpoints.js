const em = (value, base = 16) => `${value / base}em`

module.exports = {
  breakpoints: {
    xxs: em(340),
    xs: em(375),
    xsOnly: em(767),
    sm: em(768),
    smBelow: em(1023),
    'sm-and-md-only': em(1439),
    md: em(1024),
    lg: em(1440)
  },
  pixelBreakpoints: {
    xs: 375,
    xsOnly: 767,
    sm: 768,
    smBelow: 1023,
    'sm-and-md-only': 1439,
    md: 1024,
    lg: 1440
  }
}
