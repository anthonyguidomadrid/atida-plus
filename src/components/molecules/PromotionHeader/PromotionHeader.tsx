import type { ComponentPropsWithoutRef, FunctionComponent } from 'react'
import { useTranslation } from 'react-i18next'

import { Button } from '~components/atoms/Button'
import { ReactComponent as ChevronLeft } from '~assets/svg/navigation-16px/ChevronLeft.svg'
import { ProductLabels } from '~components/molecules/ProductLabels'
import { ProductLabelWrapper } from '~domains/product'
import { parseHtml } from '~helpers'
import { Asset } from '~domains/contentful'
import { Image } from '~components/atoms/Image'

export type PromotionHeaderProps = {
  title?: string
  description?: string
  image?: Asset
  labels?: ProductLabelWrapper[]
  backFunction: () => void
}

export const PromotionHeader: FunctionComponent<
  ComponentPropsWithoutRef<'div'> & PromotionHeaderProps
> = ({ title, description, image, labels, backFunction }) => {
  const { t } = useTranslation()

  return (
    <div
      className="flex flex-col sm:flex-row justify-between h-full"
      data-testid="promotionHeader"
    >
      <div className="w-full sm:col-start-1 sm:col-end-6 md:col-end-7 py-2 sm:py-3">
        <Button
          variant="back"
          icon={<ChevronLeft role="presentation" className="icon-16" />}
          onClick={backFunction}
          data-testid="promotionBackButton"
          className="mb-3 md:mb-4"
        >
          {t('shared.back')}
        </Button>
        {labels && (
          <ProductLabels
            labels={labels}
            className="flex pb-2 w-fit"
            listItemClassName="mr-1"
          />
        )}
        {title && <h1 className="text-6xl lg:text-8xl mb-1">{title}</h1>}
        {description && (
          <div>
            {parseHtml(description, {
              ul: {
                className: 'list-disc my-2 mr-2'
              },
              li: {
                className: 'ml-2'
              },
              ol: {
                className: 'list-decimal my-2 mr-2'
              }
            })}
          </div>
        )}
      </div>
      {image?.url && (
        <Image
          className="w-full h-full object-cover sm:relative"
          src={image.url}
          alt={image.title}
          loading="eager"
          aria-hidden={!image.title?.length}
        />
      )}
    </div>
  )
}
