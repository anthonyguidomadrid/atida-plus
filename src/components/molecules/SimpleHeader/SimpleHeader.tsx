import { ComponentPropsWithRef, FunctionComponent, memo } from 'react'
import classNames from 'classnames'
import { ReactComponent as Back } from '~assets/svg/navigation-24px/Back.svg'
import { Button } from '~components/atoms/Button'
import { Image } from '~components/atoms/Image'
import { Asset } from '~domains/contentful'
import {
  NavigationBreadcrumbs,
  NavigationBreadcrumb
} from '../NavigationBreadcrumbs'
import { useFeatureFlag } from '~components/helpers/FeatureFlags'
import { FeatureFlag } from '~config/constants/feature-flags'
import { useBreakpoint } from '~domains/breakpoints'
import { breakpoints } from '~domains/breakpoints/config'

export type SimpleHeaderProps = {
  backFunction: () => void
  image?: Asset
  title?: string
  isFavourites?: boolean
  breadcrumbs?: NavigationBreadcrumb[]
}

const SimpleHeaderComponent: FunctionComponent<
  ComponentPropsWithRef<'div'> & SimpleHeaderProps
> = ({ image, backFunction, title, isFavourites, breadcrumbs }) => {
  const isDesktop = useBreakpoint(breakpoints.sm)

  const isBreadcrumbsEnabled = useFeatureFlag(
    FeatureFlag.NAVIGATION_BREADCRUMBS_ON_POP_AND_COP
  )

  return (
    <div
      className={classNames('flex flex-col sm:flex-row justify-between', {
        'h-full': isFavourites,
        'h-full sm:h-22.5 md:h-24.5 lg:h-28.5': !isFavourites
      })}
      data-testid="simpleHeader"
    >
      <div
        className={classNames({
          'sm:pr-3': image?.url,
          'sm:w-full': !image?.url
        })}
      >
        <div className="flex pt-2 sm:pt-3 sm:absolute z-10">
          <Button
            variant="back"
            icon={
              <Back role="presentation" className="icon-16 align-middle mr-0" />
            }
            onClick={backFunction}
            data-testid="popBackButton"
            className={classNames('w-fit h-fit pt-3px mr-0', {
              hidden: isDesktop
            })}
          />
          {isBreadcrumbsEnabled && (
            <NavigationBreadcrumbs
              breadcrumbs={breadcrumbs ?? []}
              showOnMobile={true}
            />
          )}
        </div>
        <h1
          className={classNames('text-6xl lg:text-8xl line-clamp-2', {
            'mb-3 mt-2 sm:mt-9': isFavourites,
            'my-2 sm:my-4 lg:mt-5 sm:absolute z-10 sm:top-5.5 ': !isFavourites
          })}
        >
          {title}
        </h1>
      </div>
      {image?.url && (
        <div className=" xs-only:max-h-32.5 w-full sm:w-6/10 sm:min-w-6/10 sm:h-full lg:w-1/2 lg:min-w-1/2">
          <Image
            className="w-full h-full object-cover sm:relative"
            src={image.url}
            alt={image.title}
            loading="eager"
            preload={true}
            aria-hidden={!image.title?.length}
          />
        </div>
      )}
    </div>
  )
}

export const SimpleHeader = memo(SimpleHeaderComponent)
