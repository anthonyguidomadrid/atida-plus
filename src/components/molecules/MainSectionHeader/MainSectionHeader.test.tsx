import { render } from '@testing-library/react'
import { Provider } from 'react-redux'

import { MainSectionHeader, MainSectionHeaderProps } from '.'
import { SimpleHeader } from '~components/molecules/SimpleHeader'
import { PromotionHeader } from '~components/molecules/PromotionHeader'
import { HeroHeader } from '~components/molecules/HeroHeader'
import { CmsContentTypes } from '~config/content-types'
import { createStore } from '@reduxjs/toolkit'
import { rootReducer } from '~domains/redux'
import { setupMatchMediaMock } from '~domains/breakpoints'

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn()
  }))
})

describe(MainSectionHeader, () => {
  const defaultProps = {
    backgroundColor: 'category-beauty'
  }
  const mockHeroBanner = {
    contentType: CmsContentTypes.HERO_BANNER,
    image: {
      title: 'Some relevant image',
      alt: 'Some relevant image',
      url: 'https://picsum.photos/1000/200'
    },
    title: 'Your personal online pharmacy',
    richTextTitle: '<p>Your personal</p><p>online pharmacy</p>',
    text: 'Atida private label',
    searchPlaceholder: 'What are you looking for?',
    link: {
      label: 'view more',
      url: '/about-us'
    }
  } as const
  const setup = (
    props: MainSectionHeaderProps = defaultProps,
    Component: typeof SimpleHeader | typeof PromotionHeader = SimpleHeader
  ) =>
    render(
      <MainSectionHeader {...props}>
        {childProps => <Component {...childProps} />}
      </MainSectionHeader>
    )
  const store = createStore(rootReducer, {})

  it('applies the correct color background', async () => {
    const { findByTestId } = setup()

    expect(await findByTestId('mainSectionHeader')).toHaveClass(
      'bg-category-beauty'
    )
  })

  it('renders the SimpleHeader', async () => {
    const { findByTestId, container } = setup()

    expect(await findByTestId('simpleHeader')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('renders the PromotionHeader', async () => {
    const { findByTestId, container } = setup(defaultProps, PromotionHeader)

    expect(await findByTestId('promotionHeader')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('renders the HeroHeader with a HeroBanner', async () => {
    const { reset } = setupMatchMediaMock(true)
    const { findByTestId, container } = setup(defaultProps, () => (
      <Provider store={store}>
        <HeroHeader contentBlock={mockHeroBanner} backFunction={jest.fn()} />
      </Provider>
    ))

    expect(await findByTestId('heroHeader')).toBeInTheDocument()
    expect(await findByTestId('heroBanner')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
    reset()
  })
})
