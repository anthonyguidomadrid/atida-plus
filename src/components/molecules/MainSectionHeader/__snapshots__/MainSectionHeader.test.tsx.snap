// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSectionHeader renders the HeroHeader with a HeroBanner 1`] = `
<div>
  <header
    class="bg-category-beauty"
    data-testid="mainSectionHeader"
  >
    <div
      class="container container-fixed mx-auto sm:relative"
    >
      <div
        class="relative"
        data-testid="heroHeader"
      >
        <button
          class="button button--back absolute z-2 ml-3 sm:ml-6 lg:ml-8 top-2 sm:top-3"
          data-testid="popBackButton"
          type="button"
        >
          <svg
            class="button__icon--before icon-16"
            role="presentation"
          />
          <span
            class=""
          >
            shared.back
          </span>
        </button>
        <div
          class="w-full flex flex-col gap-5 sm:gap-6 md:gap-7 lg:gap-9"
          data-testid="contentBlocksLayout"
        >
          <div
            data-testid="heroBanner"
          />
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`MainSectionHeader renders the PromotionHeader 1`] = `
<div>
  <header
    class="bg-category-beauty"
    data-testid="mainSectionHeader"
  >
    <div
      class="container container-fixed mx-auto sm:relative"
    >
      <div
        class="flex flex-col sm:flex-row justify-between h-full"
        data-testid="promotionHeader"
      >
        <div
          class="w-full sm:col-start-1 sm:col-end-6 md:col-end-7 py-2 sm:py-3"
        >
          <button
            class="button button--back mb-3 md:mb-4"
            data-testid="promotionBackButton"
            type="button"
          >
            <svg
              class="button__icon--before icon-16"
              role="presentation"
            />
            <span
              class=""
            >
              shared.back
            </span>
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`MainSectionHeader renders the SimpleHeader 1`] = `
<div>
  <header
    class="bg-category-beauty"
    data-testid="mainSectionHeader"
  >
    <div
      class="container container-fixed mx-auto sm:relative"
    >
      <div
        class="flex flex-col sm:flex-row justify-between h-full sm:h-22.5 md:h-24.5 lg:h-28.5"
        data-testid="simpleHeader"
      >
        <div
          class="sm:w-full"
        >
          <div
            class="flex pt-2 sm:pt-3 sm:absolute z-10"
          >
            <button
              class="button button--back p-0 justify-center w-5 h-5 w-fit h-fit pt-3px mr-0"
              data-testid="popBackButton"
              type="button"
            >
              <svg
                class="button__icon--before button__icon--only icon-16 align-middle mr-0"
                role="presentation"
              />
            </button>
          </div>
          <h1
            class="text-6xl lg:text-8xl line-clamp-2 my-2 sm:my-4 lg:mt-5 sm:absolute z-10 sm:top-5.5 "
          />
        </div>
      </div>
    </div>
  </header>
</div>
`;
